Mathlib/LinearAlgebra/Dimension/Finite.lean

```diff
@@ -546,3 +546,12 @@ theorem finrank_le_one (v : M) (h : ∀ w : M, ∃ c : R, c • v = w) : finrank
   · exact (finrank_eq_one v hn h).le
 
 end RankOne
+
+namespace Module
+variable {ι : Type*}
+
+@[simp] lemma finite_finsupp [StrongRankCondition R] : Module.Finite R (ι →₀ R) ↔ Finite ι where
```

Comment:
Does this generalize to something like:
```suggestion
@[simp] lemma finite_finsupp [Module.Free R] : Module.Finite R (ι →₀ M) ↔ Module.Finite R M ∧ Finite ι where
```

Reply:
I was hoping no one would ask 🙈

Reply:
Hmm, I believe this is true (no need of freeness or rank conditions!)
```
example [Nontrivial M] : Module.Finite R (ι →₀ M) ↔ Module.Finite R M ∧ Finite ι where
```
The reason is that each generator is supported on finitely many elements of `ι`, so if there are only finitely many generators then `ι` is finite.

Reply:
Good point, I'll try removing the strong rank condition entirely tomorrow

Reply:
Done!